src/models/storage/frame.py

```diff
@@ -47,3 +47,6 @@ def __eq__(self, other):
         return self.index == other.index and \
             np.array_equal(self.data, other.data) and \
             self.info == other.info
+
+    def asdict(self):
+        return {'id': self._id, 'data': self._data, 'info': self._info}
```

Comment:
Need to also update the "info" into column scheme. 
There should be a better way doing this. Otherwise, every time we change the row structure (likely when adding more functionality), we need to manually sync across the code base.

Reply:
I remember removing the usage of `Frame`. It was used in the old version of `Batch` implementation.

The purpose of `pd.DataFrame` based `Batch` implementation was extensibility and supporting different row structures.

Reply:
I guess Frame is only being used in some tests and as a param typing hint. I was supposed to remove it from the code base when I changed the `Batch` implementation but left it as such in view of time. Feel free to delete this part.

Reply:
Removed